### PR TITLE
add ListTags TagResource UntagResourc to Lambda

### PR DESCRIPTION
**Pull request by noam-stein**
This is a fix for: [#1116](https://github.com/localstack/localstack/issues/1116)
add support for:
* list-tags
* tag-resource
* untag-resource

in lambda

### CODE REVIEW

File: localstack/services/awslambda/lambda_api.py [hunk 1]
```diff
@@ -1060,6 +1061,50 @@ def put_concurrency(version, function):
     return jsonify(data)
 
 
+@app.route('/<version>/tags/<arn>', methods=['GET'])
+def list_tags(version, arn):
+    # the version for put_concurrency != PATH_ROOT, at the time of this
```
Comment (whummer) [inline localstack/services/awslambda/lambda_api.py]: I think we can remove these 2 lines of comments (same for the other functions)

File: localstack/services/awslambda/lambda_api.py [hunk 2]
```diff
@@ -1060,6 +1061,50 @@ def put_concurrency(version, function):
...
+    return jsonify(result)
+
+
+@app.route('/<version>/tags/<arn>', methods=['POST'])
+def tag_resource(version, arn):
+    # the version for put_concurrency != PATH_ROOT, at the time of this
+    # writing it's: /2017-10-31 for this endpoint
+    # https://docs.aws.amazon.com/lambda/latest/dg/API_TagResource.html
+    data = json.loads(request.data)
+    if data.get('Tags'):
```
Comment (whummer) [inline localstack/services/awslambda/lambda_api.py]: nitpick: slightly simpler:
```
tags = data.get('Tags', {})
if tags:
    ...
```

File: localstack/services/awslambda/lambda_api.py [hunk 3]
```diff
@@ -1060,6 +1061,50 @@ def put_concurrency(version, function):
...
+def untag_resource(version, arn):
+    # the version for put_concurrency != PATH_ROOT, at the time of this
+    # writing it's: /2017-10-31 for this endpoint
+    # https://docs.aws.amazon.com/lambda/latest/dg/API_UntagResource.html
+    tag_keys = request.args.getlist('tagKeys')
+    func_details = arn_to_lambda.get(arn)
+    if func_details:
+        for tag_key in tag_keys:
+            if tag_key in func_details.tags:
+                func_details.tags.pop(tag_key)
```
Comment (whummer) [inline localstack/services/awslambda/lambda_api.py]: nitpick: lines 1101-1102 could be simplified to:
```
func_details.tags.pop(tag_key, None)
```

File: localstack/services/awslambda/lambda_api.py [hunk 4]
```diff
@@ -1060,6 +1061,50 @@ def put_concurrency(version, function):
...
+
+
+@app.route('/<version>/tags/<arn>', methods=['DELETE'])
+def untag_resource(version, arn):
+    # the version for put_concurrency != PATH_ROOT, at the time of this
+    # writing it's: /2017-10-31 for this endpoint
+    # https://docs.aws.amazon.com/lambda/latest/dg/API_UntagResource.html
+    tag_keys = request.args.getlist('tagKeys')
+    func_details = arn_to_lambda.get(arn)
+    if func_details:
```
Comment (whummer) [inline localstack/services/awslambda/lambda_api.py]: nit: I'd follow the same pattern as above:
```
if not func_details:
    return error_response('Function not found: %s' % arn, 404, error_type='ResourceNotFoundException')
...
```

File: localstack/services/awslambda/lambda_api.py (hunk 1 above)
Comment (noam-stein) [reply localstack/services/awslambda/lambda_api.py]: fixed

File: localstack/services/awslambda/lambda_api.py (hunk 4 above)
Comment (noam-stein) [reply localstack/services/awslambda/lambda_api.py]: fixed

File: localstack/services/awslambda/lambda_api.py (hunk 3 above)
Comment (noam-stein) [reply localstack/services/awslambda/lambda_api.py]: fixed

**Comment by whummer** [comment]
Great - thanks for the quick turnaround!

**Pull request merged**